pkg/syncer/syncer.go

```diff
@@ -407,7 +411,71 @@ func StartSyncer(ctx context.Context, cfg *SyncerConfig, numSyncerThreads int, i
 
 	// Start tunneler for POD access
 	if kcpfeatures.DefaultFeatureGate.Enabled(kcpfeatures.SyncerTunnel) {
-		go startSyncerTunnel(ctx, upstreamConfig, downstreamConfig, logicalcluster.From(syncTarget), cfg.SyncTargetName)
+		filter := func(http *http.Request) bool {
+			// Allow only requests prefixed by /api/v1/namespaces/.
+			if !strings.HasPrefix(http.URL.Path, "/api/v1/namespaces/") {
+				return false
+			}
+
+			// Ensure that requests are only for pods in a namespace owned by this syncer, if not, return false.
+			nsInformer, err := ddsifForDownstream.ForResource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"})
```

Comment:
What's the implication of re-creating the informer(s) every time we hit this handler? Why not just pass them in?

Reply:
I expect the ddsif to return an already instanciated informer (if it exists), but I will  pass the informers from outside as seems nicer and more correct! thanks

Reply:
done, PTAL

Reply:
It turns out that the ForResource() method of the ddsif shouldn't be used for other types than namespaces/secrets (as those informers are always available and synced), and in some cases, for ex. pods, using it, can cause the creation of a non-synced informer. Updated the code with the proper way of requesting an Informer.

Reply:
You can call `ddsifForDownstream.Start(nil)` after you call `ForResource` and it will start all not-yet-started informers.

Reply:
> You can call ddsifForDownstream.Start(nil) after you call ForResource and it will start all not-yet-started informers.

Sure, but that's not how it is designed and used in the Syncer. We want to stick to the list of informers that were created from the GVRSource (== the resources on the SyncTarget status), to avoid trying to Sync an informer that will not even find the resource types.
That's why in most places in the Syncer we use the `ddsif.Informers()` method that returns the map of synced informers.